src/ML-POD/fitpod_command.cpp

```diff
@@ -0,0 +1,1617 @@
+    memory->destroy(testdata.force);
+    memory->destroy(testdata.atomtype);
+  }
+
+  // deallocate descriptors
+
+  memory->destroy(desc.gd);
+  memory->destroy(desc.gdd);
+  memory->destroy(desc.A);
+  memory->destroy(desc.b);
+  memory->destroy(desc.c);
+  memory->destroy(desc.tmpint);
+
+  // deallocate neighbor data
+
+  memory->destroy(nb.alist);
+  memory->destroy(nb.pairnum);
+  memory->destroy(nb.pairnum_cumsum);
+  memory->destroy(nb.pairlist);
+  memory->destroy(nb.y);
+}
+
+/* ---------------------------------------------------------------------- */
+
+void CFITPOD::read_data_file(double *fitting_weights, std::string &file_format,
+                             std::string &file_extension, std::string &test_path,
+                             std::string &training_path, const std::string &data_file)
+{
+  std::string datafilename = data_file;
+  FILE *fpdata;
+  //if (comm->me == 0){
+
+    fpdata = utils::open_potential(datafilename,lmp,nullptr);
+    if (fpdata == nullptr)
+      error->one(FLERR,"Cannot open training data file {}: ",
+                                   datafilename, utils::getsyserror());
+  //}
+
+  // loop through lines of training data file and parse keywords
+
+  char line[MAXLINE],*ptr;
+  int eof = 0;
+  while (true) {
+    //if (comm->me == 0) {
+      ptr = fgets(line,MAXLINE,fpdata);
+      if (ptr == nullptr) {
+        eof = 1;
+        fclose(fpdata);
+      }
+    //}
+    //MPI_Bcast(&eof,1,MPI_INT,0,world);
+    if (eof) break;
+    //MPI_Bcast(line,MAXLINE,MPI_CHAR,0,world);
+
+    // words = ptrs to all words in line
+    // strip single and double quotes from words
+
+    std::vector<std::string> words;
+    try {
+      words = Tokenizer(utils::trim_comment(line),"\"' \t\n\r\f").as_vector();
+    } catch (TokenizerException &) {
+      // ignore
+    }
+
+    if (words.size() == 0) continue;
+
+    auto keywd = words[0];
+
+    if (words.size() != 2)
+      error->one(FLERR,"Improper POD file.", utils::getsyserror());
+
+    // settings for fitting weights
+
+    if (keywd == "fitting_weight_energy") fitting_weights[0] = utils::numeric(FLERR,words[1],false,lmp);
+    if (keywd == "fitting_weight_force") fitting_weights[1] = utils::numeric(FLERR,words[1],false,lmp);
+    if (keywd == "fitting_weight_stress") fitting_weights[2] = utils::numeric(FLERR,words[1],false,lmp);
+    if (keywd == "error_analysis_for_training_data_set") fitting_weights[3] = utils::numeric(FLERR,words[1],false,lmp);
+    if (keywd == "error_analysis_for_test_data_set") fitting_weights[4] = utils::numeric(FLERR,words[1],false,lmp);
+    if (keywd == "energy_force_calculation_for_training_data_set") fitting_weights[5] = utils::numeric(FLERR,words[1],false,lmp);
+    if (keywd == "energy_force_calculation_for_test_data_set") fitting_weights[6] = utils::numeric(FLERR,words[1],false,lmp);
+    if (keywd == "percentage_training_data_set") fitting_weights[7] = utils::numeric(FLERR,words[1],false,lmp);
```

Comment:
This seems to be a *fraction* and not a *percentage*. See my other comment.

Reply:
_percentage_ is replaced with _fraction_ in all source files.